modules/basicfuncs/list-funcs.c

```diff
@@ -70,6 +72,278 @@ tf_list_append(LogMessage *msg, gint argc, GString *argv[], GString *result)
 
 TEMPLATE_FUNCTION_SIMPLE(tf_list_append);
 
+typedef struct _LSPattern
+{
+  GString *keyword;
+  GPatternSpec *glob;
+  GString *keyword_reversed;
+  pcre *regex;
+  pcre_extra *regex_extra;
+} LSPattern;
+
+enum ListSearchMode
+{
+  LSM_LITERAL = 0,
```

Comment:
It feels like you reimplement the `LogMatcher` just working with list instead of `LogMessage`.
[optional] would you take a look if the two could have the same root ?

Reply:
I agree. I will work on the refactor by a new PR.